[feat] Implement Debug for tauri::State

### Describe the problem

I am writing an application, and I would like to add the `#[instrument]` macro from the [`tracing`](url) crate (as well as Sentry) to some of my `tauri::command`s.

However in order to do that, for some of my commands which access state, I am receiving the following error, which states that `tauri::State` does not implement `Debug`

```
error[E0277]: `tauri::State<'_, state::InfraState>` doesn't implement `std::fmt::Debug`
   --> src/ssh/port_forward.rs:722:1
    |
722 | #[instrument]
    | ^^^^^^^^^^^^^ `tauri::State<'_, state::InfraState>` cannot be formatted using `{:?}` because it doesn't implement `std::fmt::Debug`
    |
    = help: the trait `std::fmt::Debug` is not implemented for `tauri::State<'_, state::InfraState>`
    = note: required because of the requirements on the impl of `std::fmt::Debug` for `&tauri::State<'_, state::InfraState>`
note: required by a bound in `tracing::field::debug
```

### Describe the solution you'd like

Implement `Debug` for `tauri::State`

Please let me know what method works the best, I was thinking that you could utilize the `inner` method of `State`

```rust
impl<T: Send + Sync + Debug + 'static> Debug for State<'_, T> {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        write!("tauri::State \{ {} \}", self.inner().fmt())
    }
}
```

### Alternatives considered

_No response_

### Additional context

_No response_

--- a/.changes/state-debug.md
+++ b/.changes/state-debug.md
@@ -1,0 +1,5 @@
+---
+"tauri": patch
+---
+
+Implement `Debug` for `tauri::State`.

--- a/core/tauri/src/state.rs
+++ b/core/tauri/src/state.rs
@@ -37,6 +37,14 @@ impl<T: Send + Sync + 'static> Clone for State<'_, T> {
   }
 }
 
+impl<T: Send + Sync + Debug + 'static> Debug for State<'_, T> {
+  fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), Error> {
+    f.debug_tuple("State")
+      .field(&self.0)
+      .finish()
+  }
+}
+
 impl<'r, 'de: 'r, T: Send + Sync + 'static, R: Runtime> CommandArg<'de, R> for State<'r, T> {
   /// Grabs the [`State`] from the [`CommandItem`]. This will never fail.
   fn from_command(command: CommandItem<'de, R>) -> Result<Self, InvokeError> {

--- a/core/tauri/src/state.rs
+++ b/core/tauri/src/state.rs
@@ -37,11 +37,9 @@ impl<T: Send + Sync + 'static> Clone for State<'_, T> {
   }
 }
 
-impl<T: Send + Sync + Debug + 'static> Debug for State<'_, T> {
-  fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), Error> {
-    f.debug_tuple("State")
-      .field(&self.0)
-      .finish()
+impl<'r, T: Send + Sync + std::fmt::Debug> std::fmt::Debug for State<'r, T> {
+  fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    f.debug_tuple("State").field(&self.0).finish()
   }
 }
 

